extensions/amp-access/0.1/amp-access.js

```diff
@@ -142,174 +95,136 @@ export class AccessService {
-   * @private
+   * Returns the promise that will yield the access READER_ID.
+   *
+   * This is a restricted API.
+   *
+   * @return {?Promise<string>}
    */
-  onDomUpdate_(event) {
-    // Only re-authorize sections if the response is already available.
-    // Otherwise, just wait for the authorization - it will cover new sections.
-    // But wait for the last authorization operation to complete.
-    const response = this.authResponse_;
-    if (response) {
-      return this.lastAuthorizationPromise_.then(() => {
-        const target = dev().assertElement(event.target);
-        this.applyAuthorizationToRoot_(target,
-            /** @type {!JsonObject} */ (response));
-      });
+  getAccessReaderId() {
+    if (!this.enabled_) {
+      return null;
     }
+    return this.getReaderId_();
   }
 
   /**
-   * @param {string} name
-   * @param {!./access-vendor.AccessVendor} vendor
+   * @return {!Promise<string>}
+   * @private
    */
-  registerVendor(name, vendor) {
-    user().assert(this.type_ == AccessType.VENDOR,
-        'Acccess vendor "%s" can only be used for "type=vendor"', name);
-    const vendorAdapter = /** @type {!AccessVendorAdapter} */ (this.adapter_);
-    vendorAdapter.registerVendor(name, vendor);
+  getReaderId_() {
+    if (!this.readerIdPromise_) {
+      // No consent - an essential part of the access system.
+      const consent = Promise.resolve();
+      this.readerIdPromise_ = this.cid_.then(cid => {
+        return cid.get({scope: 'amp-access', createCookieIfNotPresent: true},
+            consent);
+      });
+    }
+    return this.readerIdPromise_;
   }
 
   /**
-   * @param {!JsonObject} configJson
-   * @return {!AccessTypeAdapterDef}
+   * @return {!Array<AccessSource>}
    * @private
    */
-  createAdapter_(configJson) {
-    const context = /** @type {!AccessTypeAdapterContextDef} */ ({
-      buildUrl: this.buildUrl.bind(this),
-      collectUrlVars: this.collectUrlVars_.bind(this),
+  parseConfig_() {
+    const rawContent = tryParseJson(this.accessElement_.textContent, e => {
+      throw user().createError('Failed to parse "amp-access" JSON: ' + e);
     });
-    const isJwt = (this.isJwtEnabled_ && configJson['jwt'] === true);
-    switch (this.type_) {
-      case AccessType.CLIENT:
-        if (isJwt) {
-          return new AccessServerJwtAdapter(this.ampdoc, configJson, context);
-        }
-        return new AccessClientAdapter(this.ampdoc, configJson, context);
-      case AccessType.SERVER:
-        if (isJwt) {
-          return new AccessServerJwtAdapter(this.ampdoc, configJson, context);
-        }
-        return new AccessServerAdapter(this.ampdoc, configJson, context);
-      case AccessType.VENDOR:
-        return new AccessVendorAdapter(this.ampdoc, configJson);
-      case AccessType.OTHER:
-        return new AccessOtherAdapter(this.ampdoc, configJson, context);
+
+    const configMap = {};
+    if (isArray(rawContent)) {
+      const contentArray = rawContent;
```

Comment:
Just do `const contentArray = /** @type {!Array<!JsonObject>} */ (rawContent)` and then use `contentArray.length`.

Reply:
done

Reply:
actually this did not work for me?